service/history/hsm/tree.go

```diff
@@ -112,6 +116,7 @@ func EventIDFromToken(token []byte) (int64, error) {
 // Node data should not be manipulated directly and should only be done using [MachineTransition] or
 // [Collection.Transtion] to ensure the tree tracks dirty states and update transition counts.
 type Node struct {
+	sync.RWMutex
```

Comment:
Nodes are always accessed under a lock, this layer of locking is redundant.

Reply:
Removed